torch/csrc/jit/codegen/cuda/index_idgraph_utils.cpp

```diff
@@ -151,6 +165,32 @@ IndexingParameters getGlobalIndexParameters(
       loop_indexing.loopDomains(),
       index_parameters.initial_concrete_id_index);
 
+  // Setup double buffer increment for producer case:
```

Comment:
I don't remember the double buffering part of indexing. @naoyam could you double check the double buffering portions of the changes in this PR stack.

Reply:
This is just incrementing producer index by one on the main stage of double buffer loop. We can now make the logic more explicit. 

Will do that in a follow up as the merged PR is already getting huge.

Reply:
That's right. `double_buffer_loop` is a loop that's double buffered in the current loop nest. We can just skip the rest if it's nullptr.